sample/android-sdk-sample/app/src/main/java/com/bitmark/androidsdksample/fragments/RegistrationIssuanceFragment.java

```diff
@@ -0,0 +1,187 @@
+
+        Button generateRandomFileBtn = view.findViewById(R.id.generateRandomFileBtn);
+        generateRandomFileBtn.setOnClickListener(onClickListener);
+
+        Button registerAssetBtn = view.findViewById(R.id.registerAssetBtn);
+        registerAssetBtn.setOnClickListener(onClickListener);
+
+        Button issueBitmarksBtn = view.findViewById(R.id.issueBitmarksBtn);
+        issueBitmarksBtn.setOnClickListener(onClickListener);
+
+        fileContentTextView = view.findViewById(R.id.fileContent);
+        filePathTextView = view.findViewById(R.id.filePath);
+        assetIdInputText = view.findViewById(R.id.assetId);
+        quantityInputText = view.findViewById(R.id.quantity);
+        bitmarksListView = view.findViewById(R.id.bitmarksListView);
+        View header = view.inflate(getContext(), R.layout.listview_header, null);
+        bitmarksListView.addHeaderView(header);
+        progressBar = view.findViewById(R.id.progressBar);
+
+        return view;
+    }
+
+    @Override
+    public void onDestroy() {
+        compositeDisposable.dispose();
+        super.onDestroy();
+    }
+
+    View.OnClickListener onClickListener = v -> {
+        switch (v.getId()) {
+            case R.id.generateRandomFileBtn:
+                generateRandomFileHandler();
+                break;
+            case R.id.registerAssetBtn:
+                registerAssetHandler();
+                break;
+            case R.id.issueBitmarksBtn:
+                issueBitmarksHandler();
+                break;
+        }
+    };
+
+    private void generateRandomFileHandler() {
+        String randomText = CommonUtil.randomAlphaNumeric(20);
+        FileOutputStream fos;
+
+        try {
+            Context context = getContext();
+            fos = context.openFileOutput(FILE_NAME, Context.MODE_PRIVATE);
+            fos.write(randomText.getBytes());
+            fos.close();
+
+            fileContentTextView.setText("File Content: " + randomText);
+            outputFilePath = context.getFilesDir() + "/" + FILE_NAME;
+            filePathTextView.setText("File Path: " + outputFilePath);
+
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void registerAssetHandler() {
+        if (!hasCurrentAccount(getContext())) return;
+
+        if (outputFilePath != null) {
+            String assetName = "YOUR_ASSET_NAME"; // Asset length must be less than or equal 64 characters
+            String assetFilePath = outputFilePath;
+
+            HashMap metadata = new HashMap<String, String>() {{
+                put("key1", "value1");
+                put("key2", "value2");
+            }}; // Metadata length must be less than or equal 2048 characters
+
+            progressBar.setVisibility(View.VISIBLE);
+            Single<List<RegistrationResponse.Asset>> single = registerAsset(Global.currentAccount, assetName, assetFilePath, metadata);
+            Disposable disposable = single.subscribeOn(Schedulers.io())
```

Comment:
I think it's redundant 
```.subscribeOn(Schedulers.io())
      .observeOn(AndroidSchedulers.mainThread())```

Users may alway put it in their code although we have already handled it in our SDK